### PR TITLE
Make handler configurable

### PR DESCRIPTION
**Pull request by Bl00D4NGEL**
*Description of changes:*
This PR enables the user to be able to configure the (HTTP) handler of the aws sdk.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -46,6 +46,7 @@ public function getConfigTreeBuilder()
                 ->variableNode('stats')->end()
                 ->scalarNode('endpoint')->end()
                 ->variableNode('endpoint_discovery')->end()
+                ->scalarNode('handler')->end()
```
Comment (Bl00D4NGEL) [inline src/DependencyInjection/Configuration.php:49]: I am not sure if the option should just be called "handler" or "http_handler", either option key is actually allowed to be used for the sdk. However in order to reduce confusion as to why this option isn't in the "http" arrray I went with just `handler` as a name.

**Comment by Bl00D4NGEL** [comment]
@SamRemis any chance you could have a look at this? :)

File: src/DependencyInjection/Configuration.php (hunk 1 above)
Comment (SamRemis) [reply src/DependencyInjection/Configuration.php:49]: The `handler` and the `http_handler` options are different things in the SDK.  The `handler` option takes a command object and returns a promise with the value sent back by the service.  The `http_handler` is similar, but the input is a PSR-7 request object instead.  This means the `handler` comes into play earlier and is what forms the request, the `http_handler` is used after the request is fully formed.  

Which of these functionalities are you hoping to enable?

**Comment by SamRemis** [comment]
@Bl00D4NGEL Sorry for the delay, you got me right when I was on a vacation.  If I don't respond fast enough, @stobrien89 is a good person to contact as well :) 
I asked a question in the PR; I'm thinking you'll want to change it to `http_handler` based on what I'm reading, but I'm not positive.  Thank you for the PR!

File: src/DependencyInjection/Configuration.php (hunk 1 above)
Comment (Bl00D4NGEL) [reply src/DependencyInjection/Configuration.php:49]: Ah, I see, thanks for the explanation!

In our case we would want to overwrite the http_handler, however I guess giving the option to overwrite either would probably be helpful to the consumers of this package. I can add both options in this PR unless you see that either option isn't "recommended" or anything.

**Pull request closed without merging**